src/Diagnostics.AIProjects/SearchAPI/run.pyw

```diff
@@ -0,0 +1,334 @@
+import os, gc, shutil, uuid
+from flask import Flask, request
+from RegistryReader import githubFolderPath
+from Logger import *
+from datetime import datetime, timezone
+import json, itertools, nltk
+from functools import wraps
+
+try:
+    nltk.data.find('tokenizers/punkt')
+except LookupError:
+    nltk.download('punkt')
+try:
+    nltk.data.find('stopwords')
+except LookupError:
+    nltk.download('stopwords')
+from nltk.corpus import stopwords
+from gensim.models import TfidfModel
+from nltk.stem.porter import *
+from gensim import corpora, similarities
+
+class ModelDownloadFailed(Exception):
+    pass
+class ModelFileConfigFailed(Exception):
+    pass
+class ModelFileLoadFailed(Exception):
+    pass
+class ResourceConfigDownloadFailed(Exception):
+    pass
+def getUTCTime():
+    return datetime.now(timezone.utc)
+
+def getLatency(startTime, endTime):
+    return (endTime-startTime).total_seconds()*1000
+
+packageFileNames = {
+    "dictionaryFile": "dictionary.dict",
+    "m1ModelFile": "m1.model",
+    "m1IndexFile": "m1.index",
+    "m2ModelFile": "m2.model",
+    "m2IndexFile": "m2.index",
+    "detectorsFile": "Detectors.json",
+    "sampleUtterancesFile": "SampleUtterances.json"
+}
+
+modelsPath = githubFolderPath
+def copyFolder(src, dst):
+    if os.path.isdir(src):
+        if os.path.isdir(dst):
+            shutil.rmtree(dst)
+        shutil.copytree(src, dst)
+def downloadModels(productid, path=""):
+    try:
+        copyFolder(os.path.join(modelsPath, productid), os.path.join(path, productid))
+    except Exception as e:
+        raise ModelDownloadFailed("Model can't be downloaded " + str(e))
+def downloadResourceConfig():
+    try:
+        copyFolder(os.path.join(modelsPath, "resourceConfig"), os.path.join(os.path.dirname(os.path.abspath(__file__)), "resourceConfig"))
+    except Exception as e:
+        raise ResourceConfigDownloadFailed("Resource config can't be downloaded " + str(e))
+
+downloadResourceConfig()
+config = json.loads(open("resourceConfig/config.json", "r").read())
+resourceConfig = config["resourceConfig"]
+def getProductId(resourceObj):
+    productids = []
+    if resourceObj["ResourceType"] == "App":
```

Comment:
Does this support other resource types other than App?

Reply:
Yes, it will support all other resource types. This special handling has been done for "App" resource type since it has multiple variations.